ouroboros-consensus/src/Ouroboros/Storage/ImmutableDB/Impl/Index/Cache.hs

```diff
@@ -356,7 +354,8 @@ data CacheEnv m hash h = CacheEnv
   , tracer      :: Tracer m TraceCacheEvent
   , cacheVar    :: StrictMVar m (Cached hash)
   , cacheConfig :: CacheConfig
-  , bgThreadVar :: StrictTMVar m (Thread m Void)
+  , bgThreadVar :: StrictMVar m (Maybe (Thread m Void))
+                -- ^ Nothing if no thread running
```

Comment:
Align on `bgThreadVar` :pray: :smile:

Reply:
Well okay then, but only because you helped debug this.